src/pages/trainer/training/edit/actions/updateEditor.ts

```diff
@@ -0,0 +1,15 @@
+import {trainerActionEnums} from '../../../../../common/actionEnums/trainer';
+import {textAreaTool} from '../../../../../common/ui/tools/textAreaTool';
+
+export const updateEditorAction = (editor: HTMLTextAreaElement, caret: string, offset: number) => {
+  const contentWithCaret = textAreaTool.insertAtCaretGetText(editor, caret, offset);
+  const cursorStartPosition = textAreaTool.caculateStartCursorPositionPlusOffset(editor, offset);
```

Comment:
I think this is UI logic it should be enclosed in side the component, and we should just pass as params of the action the fields to be stored in the reducer. What's your take on this? May I'm missing something?